### PR TITLE
Basic Team Setup

### PR DESCRIPTION
**Pull request by tonberry**
Addition of teams.  Split top into one team and bottom into another.  Can be seen using their move colors.

### CODE REVIEW

File: src/Map/TileMap.cs [hunk 1]
```diff
@@ -40,9 +40,12 @@ class TileMap
 
         private MouseProcessor mouseProcessor = new MouseProcessor();
 
-        // TODO: Replace with actual object representation, rather than just generic object
+        //Dictionary containing all entities and their positions on the map
         Dictionary<Entity, Vector2> PlacedObjects = new Dictionary<Entity, Vector2>();
 
+        //DIctionary containing all entities and the team that they belong to
```
Comment (ryzngard) [inline src/Map/TileMap.cs]: Dictionary

File: src/Map/TileMap.cs [hunk 2]
```diff
@@ -40,9 +40,12 @@ class TileMap
...
+        Dictionary<Entity, Team> TeamDictionary = new Dictionary<Entity, Team>();
```
Comment (ryzngard) [inline src/Map/TileMap.cs:47]: Yea, it seems odd to have Team assignment here. Maybe we should rethink where we store PlacedObjects. As long as the map has a reference to the Entities and where they are placed, it doesn't need to be saved here. Maybe we start storing location on entity, and have a singleton list of teams. Then the teams keep the reference to the entities on them.

File: src/Map/Team.cs [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+    {
+        /// <summary>
+        /// Name of the team
+        /// </summary>
+        public string name;
+
+        /// <summary>
+        /// Team's color.  Currently used for highlighting movement.
+        /// </summary>
+        public Color color;
```
Comment (ryzngard) [inline src/Map/Team.cs:19]: For now color is fine, but I'm not sure what it really means in the end. I don't know how to just tint assets easily, at least as well as have them look good

File: src/Map/TileMap.cs (hunk 1 above)
Comment (tonberry) [inline src/Map/TileMap.cs]: Fixed.

**Pull request merged**